src/uwtools/cli.py

```diff
@@ -290,6 +292,57 @@ def _dispatch_config_validate(args: Args) -> bool:
     )
 
 
+# Mode esg grid
+
+
+def _add_subparser_esg_grid(subparsers: Subparsers) -> ModeChecks:
+    """
+    Subparser for mode: esg_grid
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    """
+    parser = _add_subparser(subparsers, STR.esggrid, "Execute Esg Grid tasks")
+    _basic_setup(parser)
+    subparsers = _add_subparsers(parser, STR.action, STR.task.upper())
+    return {
+        task: _add_subparser_esg_grid_task(subparsers, task, helpmsg)
+        for task, helpmsg in uwtools.api.esg_grid.tasks().items()
+    }
+
+
+def _add_subparser_esg_grid_task(subparsers: Subparsers, task: str, helpmsg: str) -> ActionChecks:
+    """
+    Subparser for mode: esg_grid <task>
+    :param subparsers: Parent parser's subparsers, to add this subparser to.
+    :param task: The task to add a subparser for.
+    :param helpmsg: Help message for task.
+    """
+    parser = _add_subparser(subparsers, task, helpmsg.rstrip("."))
+    required = parser.add_argument_group(TITLE_REQ_ARG)
+    _add_arg_config_file(group=required, required=True)
```

Comment:
Check the other drivers, but I think `--config-file` should be optional, with a read from `stdin` performed if no value is supplied.

Reply:
The other drivers all list this as optional. I'll update this.